api/src/main/java/org/openmrs/api/stream/StreamDataService.java

```diff
@@ -106,7 +106,6 @@ public int read() throws IOException {
 		 */
 		public void propagateStreamException(IOException streamException) {
 			this.streamException = streamException;
-			throw new UncheckedIOException(this.streamException);
```

Comment:
Logging and propagating the error to reading thread instead of rethrowing to get rid of an ugly exception in tests.